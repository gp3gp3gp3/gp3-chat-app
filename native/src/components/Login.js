import React from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

const Login = () => {
  return (
    <View style={styles.wrapper}>
      <Text
        style={styles.button}
      >Login with Google</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fafbfc'
  },
  button: {
    margin: 20,
    fontSize: 30
  }
})

export default Login
