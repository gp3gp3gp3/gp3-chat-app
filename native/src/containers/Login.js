import React from 'react'
import { connect } from 'react-redux'
import { login } from '../actions'
import Login from '../components/Login'

export const LoginContainer = props => <Login {...props} />

export default connect(null, { login })(LoginContainer)
