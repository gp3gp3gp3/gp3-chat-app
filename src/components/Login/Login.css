.Login-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  background-color: white;
  min-height: calc(100vh - 100px)
}

.Login-button {
  font-size: 2em;
  background-color: lightgrey;
  padding: 20px;
  border-radius: 5px;
  cursor: pointer;
}
