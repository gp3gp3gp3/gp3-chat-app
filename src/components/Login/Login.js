import React from 'react'
import './Login.css'

const Login = ({ login }) => {
  const onLoginClick = () => login()
  return (
    <div className='Login-wrapper'>
      <div>
        <span
          className='Login-button'
          onClick={onLoginClick}
        >Login with Google</span>
      </div>
    </div>

  )
}

export default Login
