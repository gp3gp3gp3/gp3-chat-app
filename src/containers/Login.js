import React, { Component } from 'react'
import { connect } from 'react-redux'
import { login } from '../actions'

class LoginContainer extends Component {
  render () {
    return (
      <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', backgroundColor: 'white', minHeight: 'calc(100vh - 100px)' }}>
        <div><span onClick={this.props.login} style={{fontSize: '2em', backgroundColor: 'lightGrey', padding: 20, borderRadius: 5, cursor: 'pointer'}}>Login with Google</span></div>
      </div>
    )
  }
}

export default connect(null, { login })(LoginContainer)
