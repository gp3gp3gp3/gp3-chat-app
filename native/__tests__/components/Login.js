/* eslint-env jest */
import 'react-native'
// Note: test renderer must be required after react-native.
import { shallow } from 'enzyme'
import React from 'react'
import Login from '../../src/components/Login'

it('renders correctly', () => {
  const wrapper = shallow(<Login />)
  expect(wrapper).toMatchSnapshot()
})
