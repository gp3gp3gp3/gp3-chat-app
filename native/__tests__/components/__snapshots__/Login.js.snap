// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
            "margin": 20,
          }
        }
      >
        Login with Google
      </Text>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fafbfc",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Login with Google",
        "ellipsizeMode": "tail",
        "style": Object {
          "fontSize": 30,
          "margin": 20,
        },
      },
      "ref": null,
      "rendered": "Login with Google",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "margin": 20,
            }
          }
        >
          Login with Google
        </Text>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fafbfc",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Login with Google",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 30,
            "margin": 20,
          },
        },
        "ref": null,
        "rendered": "Login with Google",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
